heartbeat/scheduler/scheduler_test.go

```diff
@@ -208,7 +210,10 @@ func TestScheduler_runRecursiveTask(t *testing.T) {
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
 			limit := int64(100)
-			s := NewWithLocation(limit, monitoring.NewRegistry(), tarawaTime())
+			s := NewWithLocation(limit, monitoring.NewRegistry(), tarawaTime(), nil)
+
+			jobSem := semaphore.NewWeighted(math.MaxInt64)
+			jobSem.Acquire(context.Background(), 1)
```

Comment:
needed to acquire as the semaphore acquire is handled by `runRecursiveJob` in main scheduler.

Reply:
Can we make this an inline comment? Future coders will be grateful!

Reply:
Removed that code path.